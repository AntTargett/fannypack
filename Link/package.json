{
  "name": "fannypack/Link",
  "private": true,
  "main": "../lib/Link/index.js",
  "module": "../es/Link/index.js"
}