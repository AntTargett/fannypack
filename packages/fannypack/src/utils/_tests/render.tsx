import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
