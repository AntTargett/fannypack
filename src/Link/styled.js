import Link from 'reakit/Link';
import styled from 'reakit/styled';
import { palette, theme } from 'styled-tools';
import { darken } from 'polished';
import Icon from '../Icon/styled';

export default styled(Link)`
  color: ${palette('primary')};
  fill: ${palette('primary')};
  cursor: pointer;
  text-decoration: underline;
  text-decoration-skip: ink edges;

  &:hover {
    color: ${props => darken(0.5, palette('primary')(props))};
    fill: ${props => darken(0.5, palette('primary')(props))};
  }

  & ${Icon} {
    top: 0.15em;
  }

  & {
    ${theme('fannypack.Link.base')};
  }
`;
