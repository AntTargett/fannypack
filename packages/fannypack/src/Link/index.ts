import Link from './Link';
import * as styledLink from './styled';

export default Link;

export * from './Link';
export { default as Link } from './Link';

export { styledLink };
