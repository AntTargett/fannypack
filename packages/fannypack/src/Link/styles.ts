import { cssClass, palette, theme } from '../styled';

export const Link = styleProps => cssClass`
  color: ${palette('primary')(styleProps)};
  fill: ${palette('primary')(styleProps)};
  cursor: pointer;
  text-decoration: underline;
  text-decoration-skip: ink edges;

  &:hover {
    color: ${palette('primary900')(styleProps)};
    fill: ${palette('primary900')(styleProps)};

    & {
      ${theme('Link.hover')(styleProps)};
    }
  }

  &:focus {
    outline-style: dashed;

    & {
      ${theme('Link.focus')(styleProps)};
    }
  }

  & {
    ${theme('Link.base')(styleProps)};
  }
`;
