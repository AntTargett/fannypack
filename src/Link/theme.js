import { css } from 'reakit/styled';
import { palette } from 'styled-tools';
import { darken } from 'polished';

export default ({ base: baseOverrides } = {}) => css`
  color: ${palette('primary')}
  cursor: pointer;
  text-decoration: underline;
  text-decoration-skip: ink edges;

  &:hover {
    color: ${props => darken(0.5, palette('primary')(props))};
  }

  & {
    ${baseOverrides};
  }
`;
