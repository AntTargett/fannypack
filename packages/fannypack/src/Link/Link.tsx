import { Box as ReakitBox } from 'reakit';

import { LinkThemeConfig } from '../types';
import { useClassName, createComponent, createElement } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalLinkProps = {
  overrides?: LinkThemeConfig;
};
export type LinkProps = BoxProps & LocalLinkProps;

function useProps(props: Partial<LinkProps> = {}) {
  const boxProps = Box.useProps(props);

  const className = useClassName({
    style: styles.Link,
    styleProps: props,
    prevClassName: boxProps.className
  });

  return { ...boxProps, className };
}

export const Link = createComponent<LinkProps>(
  props => {
    const { children, use, ...restProps } = props;
    const linkProps = useProps(restProps);
    return createElement({ children, component: ReakitBox, use, htmlProps: linkProps });
  },
  {
    attach: {
      defaultProps: {
        use: 'a'
      },
      useProps
    },
    themeKey: 'Link'
  }
);
