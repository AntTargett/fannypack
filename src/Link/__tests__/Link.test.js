import React from 'react';
import render from '../../_utils/tests/render';
import Link from '../Link';
import 'jest-styled-components';

it('renders correctly for a basic link', () => {
  const { container } = render(<Link>test</Link>);
  expect(container.firstChild).toMatchSnapshot();
});
