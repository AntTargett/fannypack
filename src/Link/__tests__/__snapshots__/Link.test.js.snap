// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic link 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: #112ebb;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #112ebb;
  outline-offset: 2px;
}

.c2 {
  color: #112ebb;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

.c2:hover {
  color: #000;
}

<div
  class="c0 c1"
>
  <a
    class="c2 c3"
  >
    test
  </a>
</div>
`;
